### ISSUE TITLE
Dreambooth training producing worse results on revision c779fec04

### ISSUE THREAD

**Issue by lolcharacters** [d8ahazard/sd_dreambooth_extension#763]
```
Python 3.10.6
Commit hash: 5f4fa942b8ec3ed
Installing requirements for Web UI
Skipping dreambooth installation.
Dreambooth revision is c5cb58328c555ac276794
Diffusers version is 0.10.2
Torch version is 1.12.1
Torch vision version is 0.13.1
```

**Have you read the [Readme](https://github.com/d8ahazard/sd_dreambooth_extension#readme)?**
yes
**Have you completely restarted the stable-diffusion-webUI, not just reloaded the UI?**
yes
**Have you updated the Stable-Diffusion-WebUI to the latest version?**
no

**Describe the bug**

I've been training on an older version (c5cb58328c555ac276794) for awhile and decide to update to c779fec04. I've put identical settings and the sample images are always varying between good images and bad images. See https://imgur.com/a/ABWq3Ss where I get the first sample image and then the second image right after. In the graph, the loss rate is always dropping, so this is very strange. Also it seems like the training is overall worse. For the first 2400 steps, the newer version produces: https://imgur.com/a/SNPyYJ2 (not in order). In the older version, it produces https://imgur.com/a/m360mdv.

In addition, when I try to txt2img, (example: https://imgur.com/a/9FdLkG8). it always produces the bad images and never alternates between the good images like how it did for sampling. If I generate samples from the dreambooth section, it sometimes generates good pictures. Could this be related to the new YAML configs (#757) ?

**Environment**

What OS? Windows

If Windows - WSL or native? Native

What GPU are you using? RTX 3080

**Screenshots/Config**
If the issue is specific to an error while training, please provide a screenshot of training parameters or the
db_config.json file from /models/dreambooth/MODELNAME/db_config.json

On the newer commit: https://pastebin.com/SrBLXAUp
On the older commit: https://pastebin.com/MDWUNZw7

**Comment by ExponentialML** [comment]
Hello @lolcharacters .
There are a few anomalies I'm seeing with your training in general which may not allow it to work properly.

You have `"prior_loss_weight": 1` with `"class_prompt": "[filewords]"`, but don't have a `"class_token": ""` **_and_** `"class_data_dir": "`. 
You need *all four*, and when you have all four of these parameters set, prior preservation will work. Remember that these are pictures of the class you are trying to train (which should be "woman" or "character" in your case). 
The method you're using works, but is very experimental (works nice for specific use cases). 

Use a rare word as your instance token. As it stands now, you're using a token that is using *3 separate tokens*, so it's treating it as three separate words. 
Instead of _"kai'sa"_, do **"shld"**, which is a rare token that also counts as just one word (so your subject will be associated with it).  

You can leave the `"instance_token":` field blank. Just do `"instance_prompt": "shld, filewords"`.

You have no class directory, which means `prior preserveration` isn't working properly. If you want to fine tune the model *without* `prior preservation`, change  `"prior_loss_weight"` to `0`, and leave any class fields blank. Just train with your `"instance_data_dir": "your files with [filewords]" `

Please try the following settings below and report back with your results (just include your fine tuning optimizations like xformers). 
I also recommend setting `Clip Skip: 2` as you're doing tag based prompts, but do this after testing just `1` to see if it improves results.

First, create a new model then restart your webui, refresh, then move towards training.

`Training Steps Per Image (Epochs): 200`
`Learning Rate Scheduler: "constant_with_warmup". `
`Learning Rate Warmup Steps: 0`
`Learning Rate: 5e-6 (We're setting this a little bit higher for testing purposes, and should also work well for your case in general).`
`Clip Skip: 2`
`Prior Loss Weight: 0`
`Instance Prompt: shld, [filewords]`
`Sample Image Prompt: shld, [filewords]`

**Comment by lolcharacters** [comment]
Makes sense, I'm looking through the old code and it just checks if the # of class images is > 0. Looks like the newer code has more checks I think. I'll try updating everything and see if I'm able to replicate my previous character models.

**Comment by AyoKeito** [comment]
I have something even worse with all the new versions. Tried clean install, nothing changes.
Training 6 concepts (styles).
All results produced today, just on different versions of webui+dreambooth. Same dataset, no class images.
[Old version result](http://c2n.me/4hzOvG8.png) (6000 steps) vs [new result](http://c2n.me/4hzOx6o.png) (4500 steps) and [45000 steps.](http://c2n.me/4hzOzTD.png)
[Old version config](https://pastebin.com/fqgxrWxE) vs [new version config](https://pastebin.com/2RPkXhU8) (i've tried the new one with lora just to get the result faster after setting prior_loss_weight to 0, it's still broken even without lora).
Not sure what to do other than stick to the old version.

Version that works:

> Python revision: 3.10.8 (tags/v3.10.8:aaaf517, Oct 11 2022, 16:50:30) [MSC v.1933 64 bit (AMD64)]
Dreambooth revision: c5cb58328c555ac27679422b1da940a9b19de6f2
SD-WebUI revision: 4af3ca5393151d61363c30eef4965e694eeac15e
[+] bitsandbytes version 0.35.0 installed.
[+] diffusers version 0.10.2 installed.
[+] transformers version 4.25.1 installed.
[+] xformers version 0.0.14.dev0 installed.
[+] torch version 1.12.1+cu116 installed.
[+] torchvision version 0.13.1+cu116 installed.

Version that doesn't (clean install):

> Python revision: 3.10.8 (tags/v3.10.8:aaaf517, Oct 11 2022, 16:50:30) [MSC v.1933 64 bit (AMD64)]
Dreambooth revision: 17c3864803ebb50615205271de687be96cfc96e8
SD-WebUI revision: d7aec59c4eb02f723b3d55c6f927a42e97acd679
[+] bitsandbytes version 0.35.0 installed.
[+] diffusers version 0.10.2 installed.
[+] transformers version 4.25.1 installed.
[+] xformers version 0.0.14.dev0 installed.
[+] torch version 1.12.1+cu113 installed.
[+] torchvision version 0.13.1+cu113 installed.

Native Windows 10, NVIDIA 4090, latest drivers.

**Comment by leandro-gracia-gil** [comment]
I'm also having similar issues, though in my case the most recent dreambooth commit that works is [fab41d85](https://github.com/d8ahazard/sd_dreambooth_extension/commit/fab41d85da47bc2901a1db712e8ae265306c93b9), which is a bit more recent.

After that, the problem is that the extension is broken (you hit various errors if you try to train) and multiple branches are spawned in such a way that fixes are not merged together until later on, but at that point result quality has already degraded. For example, commit [15244fef](https://github.com/d8ahazard/sd_dreambooth_extension/commit/15244fef18c35864be6f6c2e1c0237dadc5415c1) is not broken if you enable caching latents, but quality problems are already in. Though I don't know if this is the oldest one where this happens.

I've also noticed that a few settings now have opposite default values, including "Apply Horizontal Flip" now being disabled, "Shuffle Tags" now being enabled, and "Don't cache latents" becoming "Cache latents" with both being enabled. However, even after restoring the previous default settings for these the quality is still clearly worse than before.

**Comment by poko3347** [comment]
I have the same problem. Even when I make the settings as identical as possible, the results are significantly inferior.
To be more specific, the reproduction of character traits was considerably reduced.
I'm using 633ca33 because I don't know how to fix it. However, using this commit has resulted in an error in the latest WebUI.

**Comment by master131** [comment]
Having the same issue; reverting to https://github.com/d8ahazard/sd_dreambooth_extension/commit/fab41d85da47bc2901a1db712e8ae265306c93b9 with webui commit https://github.com/AUTOMATIC1111/stable-diffusion-webui/commit/82725f0ac439f7e3b67858d55900e95330bbd326 fixes the issue for me.

**Comment by ye7iaserag** [comment]
I think you guys need to see this https://www.youtube.com/watch?v=usgqmQ0Mq7g

**Comment by maximus-sallam** [comment]
Thank you @master131
